iso15118/evcc/controller/simulator.py

```diff
@@ -177,18 +178,13 @@ async def select_energy_service_v20(
         self, services: List[MatchedService]
     ) -> SelectedEnergyService:
         """Overrides EVControllerInterface.select_energy_service_v20()."""
-        matched_energy_services = [
-            service for service in services if service.is_energy_service
-        ]
-        if matched_energy_services:
-            top_of_list: MatchedService = services[0]
-            selected_service = SelectedEnergyService(
-                service=top_of_list.service,
-                is_free=top_of_list.is_free,
-                parameter_set=top_of_list.parameter_sets[0],
-            )
-            return selected_service
-        return None
+        top_of_list: MatchedService = services[0]
```

Comment:
Not sure if I'm missing something here. If there are multiple services offered by the secc and if the top service isn't energy service, wouldn't it fail after ServiceSelectionReq is received on the SECC side here: 
https://github.com/SwitchEV/iso15118/blob/159c6fc6e015c71d2744c044b877d97c3deba913/iso15118/secc/states/iso15118_20_states.py#L754

Reply:
but the condition that was there before wouldnt do anything useful. before we call this `select_energy_service_v20` method, we already have a guarantee that we have a matched service, so creating a list with just the matched energy services isnt anything new.

nevertheless, you are right, the first service of the SECC may not be an energy one and this fails, but anyway that is something to fix in another PR, I believe